sdk/lib/io/file_impl.dart

```diff
@@ -203,16 +203,18 @@ class _FileStreamConsumer extends StreamConsumer<List<int>> {
 
 // Class for encapsulating the native implementation of files.
 class _File extends FileSystemEntity implements File {
-  final String path;
+  Uint8List rawPath;
 
-  // Constructor for file.
-  _File(this.path) {
+  _File(String path) {
     if (path is! String) {
-      throw new ArgumentError('${Error.safeToString(path)} '
-          'is not a String');
+      throw new ArgumentError(
+          '${Error.safeToString(path)} is not a String');
     }
+    this.rawPath = new Uint8List.fromList(utf8.encode(path));
   }
 
+  _File.fromRawPath(this.rawPath) {}
```

Comment:
{} not needed. This can be just:
_File.fromRawPath(this.rawPath);

Reply:
neat! done